### ISSUE TITLE
could this work like this:

### ISSUE THREAD

**Issue by ccfredbaum** [ClubCloudTech/myvideoroom-wordpress-plugin#82]
could this work like this:

```
$security_dao = Factory::get_instance( SecurityVideoPreferenceDAO::class );

// first get the user settings
$preference = $security_dao->get_by_id( $owner_id, $room_name );

// if the user does not have any a preference object, then fall back to site override
if ( ! $preference ) {
    $preference = $security_dao->get_by_id( 
                                   SiteDefaults::USER_ID_SITE_DEFAULTS, 
                                   SiteDefaults::ROOM_NAME_SITE_DEFAULT 
                                 );
}

// if there is no user preference, and no site override then fall back to default permissions
if ( ! $preference ) {
    $preference = $security_dao->get_by_id( 
                                   SiteDefaults::USER_ID_SITE_DEFAULTS, 
                                   SiteDefaults::PERMISSIONS_TABLE 
                                 );
}

// handle if preference is still not found.
if ( ! $preference ) {
     return null; // (or create a permissions object manually with the correct default settings)
}

// at this point you are guaranteed to have a $preference object
```

_Originally posted by @alecsammon in https://github.com/ClubCloudTech/myvideoroom-wordpress-plugin/pull/81#discussion_r639572306_

**Comment by ccfredbaum** [comment]
Yeah - I thought of a function to break it out. As it stands of the three main filters so far done, one uses custom logic, and two use this one. I figured it was a 5050 call to break it to a separate function given we only have three filters by default in core (likely all other filters will be in modules) - so I opted to just update the logic as per your recommendation and leave the functions duplicated

**Comment by alecsammon** [comment]
Merged into v3.0.0
